feat(curriculum): add cipher project to Python curriculum (#52373)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-string-manipulation-by-building-a-cipher/65525e359ca28d938baa82c5.md

---
id: 65525e359ca28d938baa82c5
title: Step 27
challengeType: 20
dashedName: step-27
---

# --description--

> The addition assignment operator, `+=` has the following syntax:
> 
> ```py
> a += b
> ```
> 
> Its effect is equivalent to `a = a + b`. So it enables you to add something to the value of a variable and then assign the result to that variable.
> 
> In your case, you can use it to add characters to a string variable and assign the result to that variable, building the string gradually.
> 
> Instead of assigning `alphabet[new_index]` to `new_char`, use the addition assignment operator to add it and assign it to `encrypted_text`. Then, modify the `print` call to reflect this change.

Reviewer: ````suggestion
Instead of assigning `alphabet[new_index]` to `new_char`, use the addition assignment operator to add a value and assign it at the same time to `encrypted_text`:

```python
a = 1
a += 1
print(a) # 2
a = a + 1
print(a) # 3
```

Also, modify the `print` call to reflect this change.
````

Author: Considering that comments are explained in step 47, would it be fine to use comments to show the output in the example? BTW I might have done it before.

Reviewer: Ah. If comments are not explained yet, then no.
